src/main/java/com/aws/greengrass/util/GreengrassServiceClientFactory.java

```diff
@@ -48,23 +53,29 @@ public class GreengrassServiceClientFactory {
     @Inject
     public GreengrassServiceClientFactory(DeviceConfiguration deviceConfiguration) {
         this.deviceConfiguration = deviceConfiguration;
-        deviceConfiguration.onAnyChange((what, node) -> {
+        deviceConfiguration.onAnyChange(new BatchedSubscriber((what, node) -> {
```

Comment:
I'd undo this, just clear the client when stuff changes and do the validation when someone actually asks for the client.